analyzers/src/SonarAnalyzer.Common/Rules/ParametersCorrectOrderBase.cs

```diff
@@ -18,141 +18,161 @@
-
-            if (parameterNames.Intersect(identifierNames).Any() &&
-                HasIncorrectlyOrderedParameters(argumentIdentifiers, argumentParameterMappings, parameterNames, identifierNames,
-                    analysisContext.SemanticModel))
-            {
-                // for VB the symbol does not contain the method syntax reference
-                var secondaryLocations = methodSymbol.DeclaringSyntaxReferences
-                    .Select(s => GetMethodDeclarationIdentifierLocation(s.GetSyntax()))
-                    .WhereNotNull();
+    internal void ReportIncorrectlyOrderedParameters(SonarSyntaxNodeReportingContext analysisContext,
+        MethodParameterLookupBase<TArgumentSyntax> methodParameterLookup,
+        SeparatedSyntaxList<TArgumentSyntax> argumentList,
+        Func<Location> getLocationToReport)
+    {
+        var argumentParameterMappings = methodParameterLookup.GetAllArgumentParameterMappings()
+                                                             .ToDictionary(x => x.Node, x => x.Symbol);
 
-                analysisContext.ReportIssue(SupportedDiagnostics[0].CreateDiagnostic(analysisContext.Compilation,  getLocationToReport(), secondaryLocations, methodSymbol.Name));
-            }
+        var methodSymbol = methodParameterLookup.MethodSymbol;
+        if (methodSymbol == null)
+        {
+            return;
         }
 
-        private bool HasIncorrectlyOrderedParameters(List<ArgumentIdentifier> argumentIdentifiers,
-            Dictionary<TArgumentSyntax, IParameterSymbol> argumentParameterMappings, List<string> parameterNames,
-            List<string> identifierNames, SemanticModel semanticModel)
-        {
-            for (var i = 0; i < argumentIdentifiers.Count; i++)
-            {
-                var argumentIdentifier = argumentIdentifiers[i];
-                var identifierName = argumentIdentifier.IdentifierName;
-                var parameter = argumentParameterMappings[argumentIdentifier.ArgumentSyntax];
-                var parameterName = parameter.Name;
-
-                if (string.IsNullOrEmpty(identifierName) ||
-                    !parameterNames.Contains(identifierName) ||
-                    !IdentifierWithSameNameAndTypeExists(parameter))
-                {
-                    continue;
-                }
-
-                if (argumentIdentifier is PositionalArgumentIdentifier positional &&
-                    (parameter.IsParams || identifierName == parameterName))
-                {
-                    continue;
-                }
-
-                if (argumentIdentifier is NamedArgumentIdentifier named &&
-                    (!identifierNames.Contains(named.DeclaredName) || named.DeclaredName == named.IdentifierName))
-                {
-                    continue;
-                }
-
-                return true;
-            }
+        var parameterNames = argumentParameterMappings.Values
+            .Select(x => x.Name.ToLowerInvariant())
```

Comment:
Is there a reason not to follow S4040 here and on the other places? As long as it used only for comparison, it should be fine.

Reply:
A few UTs were failing with that change, It took some time to locate the issue. Now we migrated `ToUpperCase` following what the rule suggests